### ISSUE TITLE
Process for modifing/adding css

### ISSUE THREAD

**Issue by billybonks** [babybuddy/babybuddy#758]
I deployed my mobile PR to test https://github.com/babybuddy/babybuddy/pull/754, after deployment i went to the URLS and got a 500

<img width="1111" alt="image" src="https://github.com/babybuddy/babybuddy/assets/1640136/12419f51-d51d-4f1a-acb2-a2260333a03a">

The logs also just said 500. So, I tried to get the stack trace printed out by setting the debug flag on my production. This then rendered the new pages but without assets.

So I created a commitment with the assets, and that seems to work mostly.

I went through the deployment docs multiple times and could not find anything about building assets as part of the deployment.

The commit with the update static assets, seem to be more then just what my pr changed https://github.com/babybuddy/babybuddy/pull/754/commits/e4757d83b17bf27b972ec4f4961ab4139fb6d109

Not sure why  this rendered the page and without this i got 500
https://github.com/babybuddy/babybuddy/pull/754/commits/a2f68af10e3b01522ee71222d042c6b07cf73a9f

Any advice here would be appreciated

**Comment by MrApplejuice** [comment]
I am seemingly invested enough in the observing the progress, that I am happy to chip in for this!

https://github.com/billybonks/babybuddy/pull/2

This _should_ fix the static file collection which was still biting you on your dev branch. I remembered that the static file collection was a bit weird with babybuddy because ti was orchestrated in gulp and did not use the django default paths. I think my PR should set you on the right track here.

Bottom line - from my diagnosis - was that your files were not discovered during the `collectstatic`-phase because they were residing in the non-standard `static_src` directory. However, you cannot use the standard `static` directory, because `gulp clean` will eradicate all the standard directories. Instead the PR adds a rule for the `mobile`-app to copy the files in the right place before they are grabbed by `collectstatic`.

:shrug: no idea why it was done this way, but I think this should work!

**Comment by MrApplejuice** [comment]
Oh yeah... I am out of time now, but there were still some missing image files on the `mobile`-branch. And I was not able to hand-check all the different commit IDs you linked tonight. So this is only half-tested.

~~~~~~~~~~
  File "/usr/local/lib/python3.10/dist-packages/django/template/base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/usr/local/lib/python3.10/dist-packages/django/template/base.py", line 961, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.10/dist-packages/django/templatetags/static.py", line 116, in render
    url = self.url(context)
  File "/usr/local/lib/python3.10/dist-packages/django/templatetags/static.py", line 113, in url
    return self.handle_simple(path)
  File "/usr/local/lib/python3.10/dist-packages/django/templatetags/static.py", line 129, in handle_simple
    return staticfiles_storage.url(path)
  File "/usr/local/lib/python3.10/dist-packages/django/contrib/staticfiles/storage.py", line 203, in url
    return self._url(self.stored_name, name, force)
  File "/usr/local/lib/python3.10/dist-packages/django/contrib/staticfiles/storage.py", line 182, in _url
    hashed_name = hashed_name_func(*args)
  File "/usr/local/lib/python3.10/dist-packages/django/contrib/staticfiles/storage.py", line 516, in stored_name
    raise ValueError(
ValueError: Missing staticfiles manifest entry for 'babybuddy/img/feeding.svg'
~~~~~~~~~~

Anyway, in case you want to follow my testing-procedure: I used docker to build a production-like environment and then started modifying that environment. I will attach the files I used in case they are of use to you!

**Comment by MrApplejuice** [comment]
[docker-test-code.zip](https://github.com/babybuddy/babybuddy/files/14173096/docker-test-code.zip)

**Comment by cdubz** [comment]
LOL @ "esoteric" static file collection. It was built this way 5+ years ago primarily so deployments wouldn't have to compile assets themselves. This was pre-Docker days. I'm not sure if the "extras" config approach makes sense exactly... I will try to make time to look at this... 🕵️

**Comment by cdubz** [comment]
@billybonks I'm not able to reproduce. It looks to me like you're branch has the right changes for the static files pipeline. You do need the generated assets committed and I see them there. Also if you `DEBUG=True` you definitely should see a stack trace not that custom 500 page. So it seems like maybe your deployment doesn't have what you think it has?

**Comment by billybonks** [comment]
Thanks, @MrApplejuice. I will try it out today. Thanks.

@cdubz, when I added DEBUG=True, it stopped giving the 500 🤷🏿 so I'm not sure 100% what the issue is.


Regarding the committing of those static files, ideally it you would just run the gulp build as part of the build process.

**Comment by cdubz** [comment]
@billybonks this sounds familiar... something about how the production static files system accounts for the `DEBUG` env in weird ways. Let me see if I can reproduce...

> Regarding the committing of those static files, ideally it you would just run the gulp build as part of the build process.

Yes, I know. And it did originally.... but again we're dealing with a 2017 asset build system :grin:

**Comment by cdubz** [comment]
Ok yeah this is reproducible by following the guidance here for simulating production assets: https://github.com/babybuddy/babybuddy/blob/3c54636790d7377436fd664bedcfb7ad27c06a3d/babybuddy/settings/development.py#L10-L19

This has definitely come up before and had something to do with whitenoise's asset handling or something... will try to dig in to it.

**Comment by cdubz** [comment]
Ok @billybonks this is happening because there are two static files missing. For whatever reason Django just loses it mind when that happens in production mode.

Add these two files and update the static assets again and things should be happy:

- `babybuddy/img/arrow-back-left.svg`
- `babybuddy/img/feeding.svg`

I also found after much digging around on SO a useful settings configuration for getting detailed logged even when `DEBUG=False`. I'll add that to the development settings file for use in cases like this.

**Comment by MrApplejuice** [comment]
## "esoteric"

> LOL @ "esoteric" static file collection. It was built this way 5+ years ago primarily so deployments wouldn't have to compile assets themselves. This was pre-Docker days. I'm not sure if the "extras" config approach makes sense exactly... I will try to make time to look at this... 🕵️

@cdubz  - please do not misread this. It is a build system, it works fine, I do not mean to _judge_ it. However, I remember that also for me, someone who used django in the past, it was a bit of a steep learning curve to understand that babybuddy's build system is _not_ using the native django static file collection, but a gulp-driven version of it.

I think this _could_ be addressed by adding a custom file-collector for django and implementing custom calls from there, but that would then in turn call out to javascript again to do some other stuff. An honestly, I think it is fine!

-----------------

## About collecting logs

You can also add this section of code to your settings.py-file that you are using in production to get back the error logs (I did that in the .zip that I uploaded):

~~~~~~~~~.py
import os

LOGGING = {
    "version": 1,
    "disable_existing_loggers": False,
    "handlers": {
        "console": {
            "class": "logging.StreamHandler",
        },
    },
    "root": {
        "handlers": ["console"],
        "level": "INFO",
    },
}
~~~~~~~~~

It comes from the django documentation itself (I only replaced "warning" with "info"): https://docs.djangoproject.com/en/5.0/topics/logging/#examples

**Comment by cdubz** [comment]
Yeah customizing the `LOGGING` was ultimately what I did to figure out the issue here. I added that with some notes to the development settings in ae76ef3606bddbc384055802832d2921fbd3b897 for future reference. I have definitely run in to this exact same thing before and forgot about it.

> I think this could be addressed by adding a custom file-collector for django and implementing custom calls from there, but that would then in turn call out to javascript from Python again to do some npm-stuff. An honestly, I think it is fine/ok as is right now, just something one needs to learn and keep in mind!

Yeah, I have poked around at this in the past but run in to various issues. Somehow gulp just keeps on chugging and gets the job done so I haven't ever committed to spending a lot of time on it.
